### PR TITLE
ContractModel: update quickcheck-contractmodel dependency to implement symbolic TxOuts

### PR DESCRIPTION
**Pull request by MaximilianAlgehed**
Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Formatting, PNG optimization, etc. are updated
    - [x] Important changes are reflected in changelog.d of the affected packages
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reference the ADR in the PR and reference the PR in the ADR (if revelant)
    - [x] Reviewer requested

### CODE REVIEW

**Comment by MaximilianAlgehed** [comment]
@sjoerdvisscher do you have time to look at this?

File: plutus-contract/src/Plutus/Contract/Test/ContractModel/Interface.hs [hunk 1]
```diff
@@ -483,9 +492,9 @@ assertModel s p = QCCM.assertModel s (p . fmap coerce)
 -- as it is a proxy for the predicate.
 observeChain :: ContractModel state
```
Comment (sjoerdvisscher) [inline plutus-contract/src/Plutus/Contract/Test/ContractModel/Interface.hs:493]: Maybe remove this and just export `QCCM.observe` directly?

File: plutus-contract/src/Plutus/Contract/Test/ContractModel/Interface.hs [hunk 2]
```diff
@@ -461,7 +465,12 @@ transfer w1 w2 = QCCM.transfer (CMI.walletAddress w1) (CMI.walletAddress w2)
 -- | Register the real token corresponding to a symbolic token created
 -- in `createToken`.
 registerToken :: String -> AssetClass -> SpecificationEmulatorTrace ()
-registerToken s ac = tell [(s, ac)]
+registerToken s ac = tell @[(String, Either TxOut AssetClass)] [(s, Right ac)]
+
+-- | Register the real TxOut corresponding to a symbolic TxOut created
+-- in `createTxOut`.
+registerTxOut :: String -> CardanoAPI.TxOut CardanoAPI.CtxUTxO QCCM.Era -> SpecificationEmulatorTrace ()
```
Comment (sjoerdvisscher) [inline plutus-contract/src/Plutus/Contract/Test/ContractModel/Interface.hs:472]: I would love to see this used in an example, but that doesn't have to happen in this PR.

Comment (MaximilianAlgehed) [reply plutus-contract/src/Plutus/Contract/Test/ContractModel/Interface.hs:472]: Yes, me too... But there were no good examples in `plutus-apps` for this, so my intention is to use the whole machinery in the `quickcheck-contractmodel-cooked` repo instead as there we have an example where adding this should be easy.

**Comment by MaximilianAlgehed** [comment]
@sjoerdvisscher I think we are ready to merge this.

**Pull request merged**